packet/packet.go

```diff
@@ -255,14 +257,12 @@ func (packet *Packet) GetIPv6NoCheck() *IPv6Hdr {
 
 // ParseL4ForIPv4 set L4 to start of L4 header, if L3 protocol is IPv4.
 func (packet *Packet) ParseL4ForIPv4() {
-	l3ptr := uintptr(unsafe.Pointer(packet.L3))
-	packet.L4 = unsafe.Pointer(l3ptr + uintptr((packet.GetIPv4NoCheck().VersionIhl&0x0f)<<2))
+	packet.L4 = unsafe.Pointer(uintptr(packet.unparsed()) + uintptr((packet.GetIPv4().VersionIhl&0x0f)<<2))
```

Comment:
It is wrong to use GetIPv4 instead of GetIPv4NoCheck here. Function GetIPv4 can return nil and program will crash in this place. It happens for example if arrived packet is VLAN packet.